packages/@react-spectrum/list/src/ListView.tsx

```diff
@@ -63,13 +63,13 @@ const ROW_HEIGHTS = {
   }
 };
 
-function useListLayout<T>(state: ListState<T>, density: ListViewProps<T>['density']) {
+function useListLayout<T>(state: ListState<T>, density: ListViewProps<T>['density'], overflowMode: ListViewProps<T>['overflowMode']) {
   let {scale} = useProvider();
   let collator = useCollator({usage: 'search', sensitivity: 'base'});
   let isEmpty = state.collection.size === 0;
   let layout = useMemo(() =>
     new ListLayout<T>({
-      estimatedRowHeight: ROW_HEIGHTS[density][scale],
+      estimatedRowHeight: overflowMode === 'wrap' ? null : ROW_HEIGHTS[density][scale],
```

Comment:
from devon's comment, ListLayout accepts both estimatedRowHeight as well as rowHeight
https://github.com/adobe/react-spectrum/blob/main/packages/%40react-stately/layout/src/ListLayout.ts#L20
when in overflowMode === 'wrap', we should use estimatedRowHeight as we were before
when in overflowMode === 'truncate', we should use rowHeight with the same value